### PR TITLE
Skip libraries consisting of the empty string

### PR DESCRIPTION
**Pull request by pchintalapudi**
Should fix https://github.com/JuliaLang/julia/pull/50745#issuecomment-1675879738 and https://github.com/JuliaLang/julia/issues/6184#issuecomment-1674892996, probably.

### CODE REVIEW

**Comment by gbaraldi** [comment]
LGTM

**Review by maleadt — COMMENTED** [review]
Not really familiar with the pass, but SGTM.

**Comment by pchintalapudi** [comment]
The test only fails due to https://github.com/JuliaLang/julia/issues/22318.

File: test/ccall.jl [hunk 1]
```diff
@@ -1915,3 +1915,17 @@ end
     ctest_total_const() = Val{ctest_total(1 + 2im)}()
     Core.Compiler.return_type(ctest_total_const, Tuple{}) == Val{2 + 0im}
 end
+
+const libfrobozz = Libdl.find_library("libfrobozz")
```
Comment (vtjnash) [inline test/ccall.jl:1919]:
```suggestion
const libfrobozz = ""
```

**Comment by maleadt** [comment]
This has not fixed the PkgEval issue I mentioned, see https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_date/2023-08/15/report.html

For example:

```
Failed to precompile SymPy [24249f21-da20-56a4-8eb1-6a02cf4ae2e6] to "/home/pkgeval/.julia/compiled/v1.11/SymPy/jl_uIzTuE".
julia: /source/usr/include/llvm/Support/Casting.h:578: decltype(auto) llvm::cast(From*) [with To = llvm::ConstantExpr; From = llvm::Value]: Assertion `isa<To>(Val) && "cast<Ty>() argument of incompatible type!"' failed.

[496] signal (6.-6): Aborted
in expression starting at /home/pkgeval/.julia/packages/SymPy/AswGW/src/SymPy.jl:39
gsignal at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
abort at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x7f802309140e)
__assert_fail at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
cast<llvm::ConstantExpr, llvm::Value> at /source/usr/include/llvm/Support/Casting.h:578 [inlined]
operator() at /source/src/jitlayers.cpp:1594
optimizeDLSyms at /source/src/jitlayers.cpp:2029 [inlined]
_jl_compile_codeinst at /source/src/jitlayers.cpp:227
jl_generate_fptr_impl at /source/src/jitlayers.cpp:525
jl_compile_method_internal at /source/src/gf.c:2475 [inlined]
jl_compile_method_internal at /source/src/gf.c:2363
gen_cfun_wrapper at /source/src/codegen.cpp:6079
emit_cfunction at /source/src/codegen.cpp:6716 [inlined]
emit_expr at /source/src/codegen.cpp:5547
emit_ssaval_assign at /source/src/codegen.cpp:5119
emit_stmtpos at /source/src/codegen.cpp:5354 [inlined]
emit_function at /source/src/codegen.cpp:8467
```

**Comment by gbaraldi** [comment]
I can't reproduce this failure, does pkgeval have this PR?

**Comment by maleadt** [comment]
Yes. You can check the `versioninfo()` that's part of every log, e.g.: https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_date/2023-08/15/Roots.primary.log, which shows 0b190b3526, the very commit where this Pr was merged.

**Comment by vtjnash** [comment]
doesn't seem to have fixed this case either:
```julia
julia> ccall(:test, Int, ())
julia: /home/vtjnash/julia1/src/jitlayers.cpp:1597: void JuliaOJIT::DLSymOptimizer::operator()(llvm::Module&): Assertion `cast<ConstantExpr>(libarg)->getOpcode() == Instruction::IntToPtr && "libarg should be either a global variable or a integer index!"' failed.

[2617529] signal (6.-6): Aborted
in expression starting at REPL[1]:1
gsignal at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
abort at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x7f3323207728)
__assert_fail at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
operator() at /home/vtjnash/julia1/src/jitlayers.cpp:1597
optimizeDLSyms at /home/vtjnash/julia1/src/jitlayers.cpp:2032 [inlined]

julia> code_llvm(()) do; ccall(:test, Int, ()); end
define i64 @"julia_#3_496"() #1 {
top:
  %test1 = load atomic i64 ()*, i64 ()** bitcast (void ()** @jlplt_test_499_got to i64 ()**) unordered, align 8
  %0 = call i64 %test1()
  ret i64 %0
}
```

**Pull request merged**